### ISSUE TITLE
Thrust exception running testMRFMapViewer

### ISSUE THREAD

**Issue by kgreenek** [mrfmap/mrfmap#1]
I'm super excited about this library!

However, when I run the testMRFMapViewer, it displays the window for just a second and then exits with the following exception:
`radix_sort: failed on 2nd step: cudaErrorInvalidConfiguration: invalid configuration argument`

I'm running this on Ubuntu 18.04 with an nvidia gtx 1070. I'm using Cuda version 11.

My cmake version is the latest version from here(https://cmake.org/download/), which is 3.18.0.

I did notice this warning when running cmake, but it built anyway:

```
CMake Warning (dev) in CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "mrfmap".
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Full output from ./testMRFMapViewer:

```
Trying to read file /home/kevin/src/mrfmap/mrfmap/config/coffin_world_640.yaml
[GVDBInference] Created context 0x55b7c18ed890 
[GVDBMRFMap] Initialized with:
	res::0.02
	const_sigmasq_poly::0.05, 0, 0
	const_bias_poly::0, 0, 0
	occ_thresh::0.5
	acc_thresh::0.1
	prior::0.1
	max_iters::3
	dims::4 4 3
	rows::480
	cols::640
	K::
381.361       0   319.5
      0 381.361   239.5
      0       0       1
	Polynomial Degree::3
	GVDB Config::3 3 3 3 3

  Device List:
Max. texture3D width: 16384
Max. texture3D height: 16384
Max. texture3D depth: 16384
   0. GeForce GTX 1070
   Using Device: 0, GeForce GTX 1070, Context: 0x55b7c18ed890
Starting GVDB Voxels. ver 1.11
 Creating Allocator..
 Creating Scene..
[GVDBInference] setting x and y to 4294967295, 4294967295
[GVDBInference] setting selected voxel id to 0
Creating context
[GVDBMapLikelihoodEstimator] Created context 0x55b7c1fcf7e0 
  Device List:
Max. texture3D width: 16384
Max. texture3D height: 16384
Max. texture3D depth: 16384
   0. GeForce GTX 1070
   Using Device: 0, GeForce GTX 1070, Context: 0x55b7c1fcf7e0
Starting GVDB Voxels. ver 1.11
 Creating Allocator..
 Creating Scene..
 Accuracy threshold is 0.1
[GVDBMapLikelihoodEstimator] setting x and y to 4294967295, 4294967295
Starting OctoMap thread...
Starting MRFMap thread...
Starting PangoViewer thread!
Adding channel 3
Starting loop!
Viewer initialized!
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  radix_sort: failed on 2nd step: cudaErrorInvalidConfiguration: invalid configuration argument
[1]    14107 abort (core dumped)  ./testMRFMapViewer
```

Full output from cmake:

```
cmake .. -GNinja -DCMAKE_INSTALL_PREFIX="../install"
-- The CUDA compiler identification is NVIDIA 11.0.194
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Setting build type to 'Release' as none was specified.
-- [MRFMap]:: Export name is mrfmap-targets
-- Eigen3 version 3.3.4 found in /usr/include/eigen3, but at least version 3.3.7 is required
-- [MRFMap]:: Cloning and configuring Eigen 3.3.7
-- [MRFMap]:: Eigen3 include directory used is /home/kevin/src/mrfmap/mrfmap/build/_deps/eigen-src
-- [MRFMap]:: Cloning and configuring octomap.
-- [octomap]:: Export name is mrfmap-targets
-- octomap building as Release
-- dynamicEDT3D building as Release
-- Found OpenCV: /usr (found version "3.2.0") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'yaml-cpp'
--   Found yaml-cpp, version 0.5.2
-- [MRFMap]:: Cloning and configuring codetimer.
-- [codetimer]:: Export name is mrfmap-targets
-- [codetimer]:: Pulling and configuring libcuckoo
-- [libcuckoo]:: Export name is mrfmap-targets
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- [MRFMap]:: Cloning and configuring pybind11.
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.17", minimum required is "2.7") 
-- Found PythonInterp: /usr/bin/python2.7 (found version "2.7.17") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so
-- pybind11 v2.5.dev1
-- Found CUDA: /usr/local/cuda (found version "11.0") 
-- [MRFMap]:: Cloning and configuring gvdb-voxels.
-- [GVDB]:: Export name is mrfmap-targets
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Found GLEW: /usr/include  
-- Autodetected CUDA architecture(s):  6.1
ARCH_FLAGS are -gencode;arch=compute_61,code=sm_61
kernel files found are cuda_gvdb_copydata.cu;cuda_gvdb_module.cu
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Skipping depth buffer test in raycast
-- Using depth image as buffer
-- [MRFMap]:: Cloning and configuring pangolin.
-- [Pangolin]:: Export name is mrfmap-targets
-- Found GLEW: /usr/lib/x86_64-linux-gnu/libGLEW.so
-- Eigen Found and Enabled
-- libdc1394 Found and Enabled
-- V4L Found and Enabled
-- Performing Test HAVE_FFMPEG_MAX_ANALYZE_DURATION
-- Performing Test HAVE_FFMPEG_MAX_ANALYZE_DURATION - Success
-- Performing Test HAVE_FFMPEG_AVFORMAT_ALLOC_OUTPUT_CONTEXT2
-- Performing Test HAVE_FFMPEG_AVFORMAT_ALLOC_OUTPUT_CONTEXT2 - Success
-- Performing Test HAVE_FFMPEG_AVPIXELFORMAT
-- Performing Test HAVE_FFMPEG_AVPIXELFORMAT - Success
-- ffmpeg Found and Enabled
-- OpenNI Found and Enabled
-- OpenNI2 Found and Enabled
-- libpng Found and Enabled
-- libjpeg Found and Enabled
-- libtiff Found and Enabled
-- libopenexr Found and Enabled
-- liblz4 Found and Enabled
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components: doxygen dot 
-- [MRFMap]:: Building in Release mode!
-- [MRFMap]:: Release build.
-- Autodetected CUDA architecture(s):  6.1
-- [MRFMap]:: ARCH_FLAGS are -gencode;arch=compute_61,code=sm_61
-- [MRFMap]:: current source is /home/kevin/src/mrfmap/mrfmap
-- [MRFMap]:: kernel files found are gvdbhelper.cu;gvdbkernels.cu
-- Configuring done
CMake Warning (dev) in CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "mrfmap".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "GVDBKERNELS_PTX".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in build/_deps/gvdb-src/source/gvdb_library/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "gvdb".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in build/_deps/gvdb-src/source/gvdb_library/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "gvdbPTX".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /home/kevin/src/mrfmap/mrfmap/build
```

**Comment by icoderaven** [comment]
Hi! And thanks for trying the library out! I've noticed the same issue when I tried building with CUDA 11 as well
https://github.com/NVIDIA/gvdb-voxels/issues/96

and it's currently unresolved. In the meanwhile I'd suggest trying it out with CUDA 10.1/2 - it's reasonably straightforward to use two CUDA installs within the same system (And there are scripts out there to switch the environment variables between the two - I use https://github.com/phohenecker/switch-cuda/blob/master/switch-cuda.sh)
